src/wp-includes/Requests/Transport/cURL.php

```diff
@@ -342,7 +342,7 @@ protected function setup_handle($url, $headers, $data, $options) {
 				$data = '';
 			}
 			elseif (!is_string($data)) {
-				$data = http_build_query($data, null, '&');
+				$data = http_build_query($data, '', '&');
```

Comment:
Requests is an upstream library. This change was made in https://github.com/WordPress/Requests/pull/500 so just need to have a new version of it set and added to WP.

I'd reckon that it can be removed from this PR and rely on the upstream reference being pulled in.

Reply:
I agree with @kraftbj. The fix in this file would be made in the Requests library. I know that @jrfnl has been working on PHP 8.1 compatibility fixes for it. Flagging her here.

Reply:
Thanks for the ping @hellofromtonya and yes, @kraftbj and @hellofromtonya are correct.
You can find the repo for Requests here: https://github.com/WordPress/Requests/ and all the PHP 8.1 related PRs here: https://github.com/WordPress/Requests/pulls?q=is%3Apr+php+8.1

All the Requests related fixes which were lined up in WP PR #1798, #1799 and this PR, have already been applied in Requests.

Work is under way to get it ready for a new release and we're hoping to release before the WP 5.9 RC deadline, but as this release will be a major (2.0), we're trying to still get some more changes in which could be considered breaking changes (stricter input validation) as otherwise those would need to wait until the next major.

If you'd like to get involved with Requests or want to help us get it ready for the next release, please know there is a dedicated channel in the WP Slack #core-http-api, so feel free to ask questions on how to get started etc there.

Once the release is out, an update of the WP code can be expected forthwith. A [test with the most pertinent breaking changes in Requests 2.0](https://github.com/WordPress/wordpress-develop/pull/1624) has already been done, so all that's really needed is for that PR to be updated with the latest changes from Requests itself.